unity-renderer/Assets/DCLServices/WearablesCatalogService/LambdasWearablesCatalogService.cs

```diff
@@ -0,0 +1,308 @@
+                wearablesInUseCounters.Remove(wearableId);
+            }
+        }
+
+        public void RemoveWearablesInUse(IEnumerable<string> wearablesInUseToRemove)
+        {
+            foreach (string wearableToRemove in wearablesInUseToRemove)
+            {
+                if (!wearablesInUseCounters.ContainsKey(wearableToRemove))
+                    continue;
+
+                wearablesInUseCounters[wearableToRemove]--;
+            }
+        }
+
+        public void EmbedWearables(IEnumerable<WearableItem> wearables)
+        {
+            foreach (WearableItem wearableItem in wearables)
+            {
+                WearablesCatalog[wearableItem.id] = wearableItem;
+
+                if (wearablesInUseCounters.ContainsKey(wearableItem.id))
+                    wearablesInUseCounters[wearableItem.id] = 10000; //A high value to ensure they are not removed
+            }
+        }
+
+        public void Clear()
+        {
+            WearablesCatalog.Clear();
+            wearablesInUseCounters.Clear();
+            awaitingWearableTasks.Clear();
+            pendingWearablesToRequest.Clear();
+        }
+
+        UniTask<(WearableResponse response, bool success)> ILambdaServiceConsumer<WearableResponse>.CreateRequest
+            (string endPoint, int pageSize, int pageNumber, CancellationToken cancellationToken) =>
+            lambdasService.Ref.Get<WearableResponse>(
+                "",
+                endPoint,
+                REQUESTS_TIME_OUT_SECONDS,
+                ATTEMPTS_NUMBER,
+                cancellationToken,
+                LambdaPaginatedResponseHelper.GetPageSizeParam(pageSize),
+                LambdaPaginatedResponseHelper.GetPageNumParam(pageNumber));
+
+        private async UniTask<WearableItem> SyncWearablesRequestsAsync(string newWearableId, CancellationToken ct)
+        {
+            if (!awaitingWearableTasks.TryGetValue(newWearableId, out var source))
+            {
+                pendingWearablesToRequest.Add(newWearableId);
+                awaitingWearableTasks[newWearableId] = source = new UniTaskCompletionSource<WearableItem>();
+            }
+
+            await UniTask.DelayFrame(FRAMES_TO_CHECK_FOR_SENDING_PENDING_REQUESTS, cancellationToken: ct);
+
+            WearableItem result = null;
+
+            if (pendingWearablesToRequest.Count > 0)
+            {
+                using var wearableIdsPool = PoolUtils.RentList<string>();
+                var wearableIds = wearableIdsPool.GetList();
+                wearableIds.AddRange(pendingWearablesToRequest);
+                pendingWearablesToRequest.Clear();
+
+                (WearableResponse response, bool success) serviceResponse;
+
+                try
+                {
+                    serviceResponse = await lambdasService.Ref.Get<WearableResponse>(
+                        "",
+                        WEARABLES_BY_ID_END_POINT.Replace("{wearableIdsQuery}", GetWearablesQuery(wearableIds)),
```

Comment:
Don't pass Url encoded params as an `endPoint`, there is a separate argument for them